lib/gnat/async_pub.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Gnat.AsyncPub do
+  use GenServer
+  require Logger
+
+  @default_queue_settings %{max_messages: 10_000, max_memory: 100 * 1024 * 1024}
+  @pause 25
+
+  def start_link(queue_settings \\ %{}, opts \\ []) do
+    GenServer.start_link(__MODULE__, queue_settings, opts)
+  end
+
+  @spec pub(GenServer.server, String.t, binary(), keyword()) :: :ok
+  def pub(pid, topic, message, opts \\ []) do
+    GenServer.call(pid, {:pub, topic, message, opts})
+  end
+
+  @impl GenServer
+  def init(%{connection_name: name}=queue_settings) when is_atom(name) do
+    queue_settings =
+      @default_queue_settings
+      |> Map.merge(queue_settings)
+      |> Map.put(:queue, :queue.new())
+    {:ok, queue_settings, 0}
+  end
+
+  @impl GenServer
+  def handle_call({:pub, _topic, _message, _opts}=todo, _from, state) do
+    state = update_in(state, [:queue], fn(q) -> :queue.in(todo, q) end)
+    {:reply, :ok, state, 0}
```

Comment:
I think using this `0` timeout for flow will probably work fine but I'd also suggest using `:continue` instead for these immediate actions based on what the GenServer docs say:
> _Because a message may arrive before the timeout is set, even a timeout of 0 milliseconds is not guaranteed to execute. To take another action immediately and unconditionally, use a :continue instruction._

Reply:
I actually wrote it using `:continue` and `handle_continue` initially, but then I realized that when someone does an async pub, the process would immediately become temporarily un-available until the `handle_continue` finishes.

If there are a bunch of `pub`s right at the same time, I think I would prefer to be responsive those processes and fit in the next push of messages to the connection right after them... I guess this means that it's possible for the process to be so busy responding to `pub`s that it never actually gets anything published 🤔 I think that would be pretty hard to do though.

Reply:
Seems like a fair trade-off.

We might be able to resolve that by using `cast` instead (unless you want `call` to guarantee the message got updated to the queue). Then could use the `:continue` flow.

But...I think its 6's at this point and can be addressed should that crazy usage case even come up. Totally good with implementation and just spitballing other thoughts

Reply:
:+1: the `cast` idea interesting for sure. I'll think about that one